### PR TITLE
[Kubernetes] Add kubeletstatsreceiver documentation

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Adds documentation for the kubeletstats receiver

Related to https://github.com/open-telemetry/opentelemetry.io/issues/3005

### CODE REVIEW

**Comment by cartermp** [comment]
@TylerHelmuth I am assuming that since this is in draft, you're not looking for a review right now?

**Comment by TylerHelmuth** [comment]
@cartermp you can review but since I branched it off of the k8sattributes PR I was waiting to mark ready until I could rebase.

**Comment by cartermp** [comment]
Will await that one then

**Comment by cartermp** [comment]
Alright @TylerHelmuth we're all merged up, so go ahead and rebase and then we'll review

File: content/en/docs/kubernetes/collector/components.md [hunk 1]
```diff
@@ -170,6 +173,81 @@ roleRef:
...
+
+| Deployment Pattern   | Usable                                                             |
+| -------------------- | ------------------------------------------------------------------ |
+| DaemonSet (Agent)    | Preferred                                                          |
+| Deployment (Gateway) | Yes, but will only collect metrics from the node it is deployed on |
+| Sidecar              | No                                                                 |
+
+Each Kubernetes node runs a kubelet which includes an API server. The Kubernetes
+Receiver connects to that kubelet via the API server to collect metrics about
+the node and the workloads running on the node.
```
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:186]:
```suggestion
Each Kubernetes node runs a kubelet that includes an API server. The Kubernetes
Receiver connects to that kubelet via the API server to collect metrics about
the node and the workloads running on the node.
```

File: content/en/docs/kubernetes/collector/components.md [hunk 2]
```diff
@@ -170,6 +173,81 @@ roleRef:
...
+There are different methods for authentication, but typically a service account
+is used. The service account will also need proper permissions to pull data from
+the Kubelet (see below). If you're using the
+[OpenTelemetry Collector Helm chart](../../helm/collector/) you can use the
+[`kubeletMetrics` preset](../../helm/collector/#kubelet-metrics-preset) to get
+started.
+
+By default metrics will be collected for pods and nodes, but you can configure
+the receiver to collect container and volume metrics as well. The receiver also
+allows configuring how often the metrics are collected:
```
Comment (cartermp) [inline content/en/docs/kubernetes/collector/components.md:197]:
```suggestion
By default, metrics will be collected for pods and nodes, but you can configure
the receiver to collect container and volume metrics as well. The receiver also
allows configuring how often the metrics are collected:
```

**Review by cartermp — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**